sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1143,6 +1143,8 @@ object PushDownPredicate extends Rule[LogicalPlan] with PredicateHelper {
     case _: Repartition => true
     case _: ScriptTransformation => true
     case _: Sort => true
+    case _: BatchEvalPython => true
```

Comment:
For my benefit, would you mind explain what does `canPushThrough` define? Are these nodes that a projection and/or filter can be pushed through?

Reply:
This defines the nodes that we can push filters through.